en/ros2/px4_ros2_navigation_interface.md

```diff
@@ -0,0 +1,290 @@
+# PX4 ROS 2 Navigation Interface
+
+:::warning Experimental
+At the time of writing, parts of the PX4 ROS 2 Interface Library are experimental, and hence subject to change.
+:::
+
+The [PX4 ROS 2 Interface Library](../ros2/px4_ros2_interface_lib.md) navigation interface enables developers to send their position measurements to PX4 directly from ROS 2 applications, such as a VIO system or a map matching system.
+The interface provides a layer of abstraction from PX4 and the uORB messaging framework, and introduces a few sanity checks on the requested state estimation updates sent via the interface.
+These measurements are then fused into the EKF as internal PX4 measurements would.
+
+The library provides two classes, `LocalPositionMeasurementInterface` and `GlobalPositionMeasurementInterface` which both expose a similar `update` method to provide either a local position or global position update to PX4, respectively.
+The `update` method expects a position measurement `struct` (defined below) which developers can populate with their own generated position measurements.
+
+## Example and First Test
+
+The following steps are required to get started:
+
+1. Make sure you have a working [ROS 2 setup](../ros/ros2_comm.md), with [`px4_msgs`](https://github.com/PX4/px4_msgs) in the ROS 2 workspace.
+2. Clone the repository into the workspace:
+
+   ```sh
+   cd $ros_workspace/src
+   git clone --recursive https://github.com/Auterion/px4-ros2-interface-lib
+   ```
+
+   :::note
+   To ensure compatibility, use the latest _main_ branches for PX4, _px4_msgs_ and the library.
+   See also [here](https://github.com/Auterion/px4-ros2-interface-lib#compatibility-with-px4).
+   :::
+
+3. Build the workspace:
+
+   ```sh
+   cd ..
+   colcon build
+   ```
+
+4. In a different shell, start PX4 SITL:
+
+   ```sh
+   cd $px4-autopilot
+   make px4_sitl gazebo-classic
+   ```
+
+   (here we use Gazebo-Classic, but you can use any model or simulator)
+
+5. In yet a different shell, run the micro XRCE agent (you can keep it running afterward):
+
+   ```sh
+   MicroXRCEAgent udp4 -p 8888
+   ```
+
+6. Back in the ROS 2 terminal, source the workspace you just built (in step 3) and run one of the examples which periodically sends dummy global position updates:
+
+   ```sh
+   source install/setup.bash
+   ros2 run example_global_navigation_cpp example_global_navigation_cpp
```

Comment:
Whats going on here? i.e. looks like you ar running it two times example_global_navigation_cpp ? Is this first the package to build and then the executable to run or something?

Reply:
Good catch! Exactly, the ros2 run syntax is `ros2 run <package_name> <executable_name>`

This command was currently functional, but I think it's bad practice to have the package and executable names the exact same.

I'm going to rename the executable `example_global_navigation_cpp` -> `example_global_navigation` in the implementation and the docs. This will then reflect the command used to run the mode examples.